treelib/tree.py

```diff
@@ -240,20 +240,17 @@ def filter_(node):
         return self.__get_iter(nid, level, filter_, key, reverse, dt, [])
 
     def __get_iter(self, nid, level, filter_, key, reverse, dt, is_last):
-        dt_vline, dt_line_box, dt_line_cor = dt
-
-        nid = self.root if (nid is None) else nid
-        if not self.contains(nid):
```

Comment:
this is not needed since it's already checked in `node = self[nid]`:
```
    def __getitem__(self, key):
        """Return _nodes[key]"""
        try:
            return self._nodes[key]
        except KeyError:
            raise NodeIDAbsentError("Node '%s' is not in the tree" % key)
```